pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -166,12 +166,13 @@ export namespace yaml {
 
         const kind = obj["kind"];
         const apiVersion = obj["apiVersion"];
-        if (kind == null || apiVersion == null) {
+        const meta = obj["metadata"];
+        if (kind == null || apiVersion == null || meta == null) {
```

Comment:
I think it's more accurate to do `"metadata" in obj` instead of doing the property lookup and testing for null. You'll get `undefined` anyway here if `metadata` isn't a property of `obj` - it works out here because of the non-strict equality, but `meta === null` will be false even if the property does not exist.